### DISCUSSION TITLE
Confusion - Your first sql website

### DISCUSSION THREAD

**Discussion by bever1337** [sqlpage/SQLPage#707] [Q&A] [answered]
Howdy, I'm excited about this project but confused by the initial setup. Following along with https://sql.datapage.app/your-first-sql-website/

1. I run the bash script. I already have homebrew so proceed to `brew install sqlpage` and `sqlpage`. The second command creates a sqlpage directory in my current path. That's unexpected and now my shell won't resolve the sqlpage CLI from this directory.
2. Create a project root folder, `mkdir web` and `cd web`.
3. Then launch `sqlpage.bin`. What is `sqlpage.bin`? This appears without previous instruction

The end result doesn't match the screenshot. The user has the following directory structure:

```
/
  /sqlpage
  /web
```

So I run `sqlpage` again, creating another `sqlpage` directory, but that's not right...

```
 sqlpage --version
sqlpage 0.30.1
```

Here's my speculation:

1. The `sqlpage` command should be run _after_ creating the project root.
2. The `sqlpage` CLI failed to create `sqlpage.bin`, causing my confusion about the `sqlpage` directory making the `sqlpage` CLI inaccessible

Can anyone spot a mistake I've made? Thanks for reading through

**Comment by bever1337** [comment]
Until I understand the (likely user) error, this is satisfactory:

```bash
mkdir demo
cd demo # project root
mkdir notsqlpage
cd notsqlpage
touch sqlpage.json
# edit json to `{ "web_root": "./www" }`
vim sqlpage.json
cd ../
mkdir www
cd www
touch index.sql
# copy paste user example from article
vim index.sql
cd ../
sqlpage --config-dir ./notsqlpage
```

project directory:
```
/demo
  /notsqlpage
    sqlpage.json
  /www
    index.sql
```

**Comment by lovasoa** [comment, marked as answer]
Hello and thank you very much for this feedback. The instructions were written for a Linux machine (where we distribute executables named `sqlpage.bin`) and I didn't realize they did not work on macos. I'll update the tutorial ASAP. 

thanks again!

**Comment by bever1337** [comment]
That's great, thank you for the help.

**Comment by bever1337** [comment]
Please let me know if I can help with testing

**Comment by lovasoa** [comment]
I updated the instructions, would love to know what you think ! https://sql.datapage.app/your-first-sql-website/?os=macos

**Comment by dr-pib** [comment]
When watching this [video](https://www.youtube.com/watch?v=RojtlWllOfE&list=PLTue_qIAHxActQnLn_tHWZUNXziZTeraB&index=2) at 0:28 you can see there is the SQLPage file and an SQLPage folder (0:31). However, on MacOS using the homebrew installation, I only have the SQLPage file and no folder. Is that correct? Is that a problem? 

The video says the folders are empty, but there is a json file that I do not have. Do I need to create the folders and json file?

**Reply by lovasoa** [reply]
**Hello again!** 😊  

No worries—you don’t need to create any of these files or folders for SQLPage to work. It runs just fine without them!  

However, since you’re curious, here’s what they do and when you might need them:  

- **`sqlpage/` (Configuration Folder)** – This is an optional folder where SQLPage looks for settings and customisations.  
- **`sqlpage/sqlpage.json` (Configuration File)** – This file notably lets you connect SQLPage to an existing database (PostgreSQL, MySQL, MSSQL, or SQLite). If you’re just using the default database, you don’t need it.  
- **`sqlpage/templates/` (Custom Components)** – This folder is for advanced users who want to design custom UI elements beyond the built-in ones. If you’re just starting out, you can safely ignore this.  

So, if you don’t see these files on your Mac installation, that’s totally fine! SQLPage is designed to work out of the box without them. Let me know if you have any other questions—happy to help!
